Tests/LibWeb/Text/input/css/filter-grayscale.html

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<script src="../include.js"></script>
+<style>
+    #high{
+        background-color: blue;
+        height: 30px;
+        width: 30px;
+        filter: grayscale(200%) brightness(200%) contrast(200%) invert(200%) opacity(200%) saturate(200%) sepia(200%);
```

Comment:
Most of these filter functions also have lower boundaries ("cannot be negative"), maybe add a #low as well?

Reply:
It seemed like this was handled elsewhere, but now I'm not sure anymore. I'll check later.

Reply:
Checked it with this:
```js
let elm = document.createElement('a');
elm.style.filter = 'grayscale(-50%)';
console.log(elm.style.filter);
```

Gives me '' on Chrome and Firefox, but 'grayscale(-50%)' in Ladybird.

Reply:
Firefox displays this as an invalid value in the developer console, and therefore ignores it entirely. This means we probably shouldn't clamp it, either. Maybe the right thing is to reject it while parsing?

Reply:
I don't see that happening there, what version of Firefox are you testing this in? It looks like `grayscale(0%)` and below (negative values) effectively get ignored.

Reply:
Right, I was looking at the console output, not the style tab of an element. Don't think we should look at Firefox though, it was just a comparison. I'd rather we implement the specs (which calls negative values invalid).

Reply:
Absolutely! I just don't really know where to start with the spec, and since the common behavior seems to be to clamp values during parsing and drop invalid values, I suggest we do the same.

Reply:
I guess just copy and paste some of the constraints outlined here as spec comments and implement them?
https://drafts.fxtf.org/filter-effects/#supported-filter-functions

Reply:
Yeah. For example, `grayscale()` says:

> Values of amount over 100% are allowed but UAs must clamp the values to 1.

So, you'd leave them as they are when parsing, and clamp them later. And...

> Negative values are not allowed.

So, you'd reject them while parsing (by returning null).

These constraints also apply to the result of `calc()` values, but it looks like our code doesn't allow them in filters. Once it does, `calc()` can't be bounds-checked when parsed, so it's always allowed (as long as it's the correct type) and then you'd need to clamp its value when it's used. (So if the `calc()` resolves to a negative number, you'd treat it as 0.)